src/EnergyPlus/Material.cc

```diff
@@ -2756,6 +2758,99 @@ void GetMaterialData(EnergyPlusData &state, bool &ErrorsFound) // set to true if
+    GetMaterialAddOnInput(state, ErrorsFound); // Read variable thermal and solar absorptance add-on data
+}
+
+void ReadingOneMaterialAddOn(EnergyPlusData &state, int i, bool &errorsFound)
+{
+    int IOStat; // IO Status when calling get input subroutine
+    int numAlphas;
+    int numNumbers;
+    Array1D_string alphas(7);   // character string data
+    Array1D<Real64> numbers(1); // numeric data
+    std::string_view cCurrentModuleObject{"MaterialProperty:VariableAbsorptance"};
+    // Call Input Get routine to retrieve material data
+    state.dataInputProcessing->inputProcessor->getObjectItem(state,
+                                                             cCurrentModuleObject,
+                                                             i,
+                                                             alphas,
+                                                             numAlphas,
+                                                             numbers,
+                                                             numNumbers,
+                                                             IOStat,
+                                                             state.dataIPShortCut->lNumericFieldBlanks,
+                                                             state.dataIPShortCut->lAlphaFieldBlanks,
+                                                             state.dataIPShortCut->cAlphaFieldNames,
+                                                             state.dataIPShortCut->cNumericFieldNames);
+
+    // Load the material derived type from the input data.
+    int MaterNum = UtilityRoutines::FindItemInPtrList(alphas(2), state.dataMaterial->Material);
+    if (MaterNum == 0) {
+        ShowSevereError(state,
+                        format("{}: invalid {} entered={}, must match to a valid Material name.",
+                               cCurrentModuleObject,
+                               state.dataIPShortCut->cAlphaFieldNames(1),
+                               alphas(2)));
+        errorsFound = true;
+        return;
+    }
+    auto *thisMaterial = state.dataMaterial->Material(MaterNum);
+
+    if (thisMaterial->Group != Material::MaterialGroup::RegularMaterial) {
+        ShowSevereError(
+            state,
+            format("{}: Reference Material is not appropriate type for Thermal/Solar Absorptance properties, material={}, must have regular "
+                   "properties (Thermal/Solar Absorptance)",
+                   cCurrentModuleObject,
+                   thisMaterial->Name));
+        errorsFound = true;
+        return;
+    }
+
+    Material::VariableAbsCtrlSignal controlSignal = Material::VariableAbsCtrlSignal::SurfaceTemperature; // default value
+    controlSignal = static_cast<VariableAbsCtrlSignal>(getEnumerationValue(VariableAbsCtrlSignalUC, UtilityRoutines::MakeUPPERCase(alphas(3))));
+    //    init to 0 as GetScheduleIndex returns 0 for not-found schedule or curves
+    int functionIdxThermal = 0;
```

Comment:
You may as well just use `thisMaterial->variableThermalAbsorptanceFunctionIdx` instead of the temporary here.

Reply:
right. changed it to `thisMaterial->variableThermalAbsorptanceFunctionIdx`